tests/unit/test_serialisation/test_serialisation.py

```diff
@@ -591,3 +594,339 @@ def test_serialised_model_plotting(self):
+    def test_serialise_scalar(self):
+        S = pybamm.Scalar(2.718)
+        j = Serialise.convert_symbol_to_json(S)
+        S2 = Serialise.convert_symbol_from_json(j)
+        assert isinstance(S2, pybamm.Scalar)
+        assert S2.value == pytest.approx(2.718)
+
+    def test_serialise_time(self):
+        t = pybamm.Time()
+        j = Serialise.convert_symbol_to_json(t)
+        t2 = Serialise.convert_symbol_from_json(j)
+        assert isinstance(t2, pybamm.Time)
+
+    def test_convert_symbol_to_json_with_number_and_list(self):
+        for val in (0, 3.14, -7, True):
+            out = Serialise.convert_symbol_to_json(val)
+            assert out is val or out == val
+
+        sample = [1, 2, 3, "foo", 4.5]
+        out = Serialise.convert_symbol_to_json(sample)
+        assert out is sample
+
+    def test_convert_symbol_from_json_with_primitives(self):
+        assert Serialise.convert_symbol_from_json(3.14) == 3.14
+        assert Serialise.convert_symbol_from_json(42) == 42
+        assert Serialise.convert_symbol_from_json(True) is True
+
+    def test_convert_symbol_from_json_with_none(self):
+        assert Serialise.convert_symbol_from_json(None) is None
+
+    def test_convert_symbol_from_json_unexpected_string(self):
+        with pytest.raises(ValueError, match=r"Unexpected raw string in JSON: foo"):
+            Serialise.convert_symbol_from_json("foo")
+
+    def test_numpy_array_conversion(self):
+        arr = np.array([1, 2, 3])
+        assert Serialise._json_encoder(arr) == [1, 2, 3]
+
+    def test_numpy_float_conversion(self):
+        val1 = np.float32(2.71)
+        result1 = Serialise._json_encoder(val1)
+        assert result1 == float(val1)
+        assert isinstance(result1, float)
+
+        val2 = np.float64(3.14)
+        result2 = Serialise._json_encoder(val2)
+        assert result2 == float(val2)
+        assert isinstance(result2, float)
+
+    def test_numpy_int_conversion(self):
+        val1 = np.int32(42)
+        result1 = Serialise._json_encoder(val1)
+        assert result1 == int(val1)
+        assert isinstance(result1, int)
+
+        val2 = np.int64(123)
+        result2 = Serialise._json_encoder(val2)
+        assert result2 == int(val2)
+        assert isinstance(result2, int)
+
+    def test_unsupported_type_raises(self):
+        class Dummy:
+            pass
+
+        with pytest.raises(TypeError, match="is not JSON serializable"):
+            Serialise._json_encoder(Dummy())
+
+    def test_create_symbol_key(self):
+        var1 = pybamm.Variable("x", bounds=(0, 1))
+        var2 = pybamm.Variable("x", bounds=(0, 2))
+
+        json1 = Serialise.convert_symbol_to_json(var1)
+        json2 = Serialise.convert_symbol_to_json(var2)
+
+        key1 = Serialise._create_symbol_key(json1)
+        key2 = Serialise._create_symbol_key(json2)
+
+        assert isinstance(key1, str)
+        assert isinstance(key2, str)
+        assert key1 != key2
```

Comment:
Superb, thanks for adding a test for this as well. Could you add another test, such as something like `test_variable_domain_collision_prevention` (or a better name of your liking), which initialises two variables with different domains, and similarly test that the symbol keys they create are unique? Just covering the gaps in testing.